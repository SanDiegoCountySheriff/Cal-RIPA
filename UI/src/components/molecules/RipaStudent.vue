<template>
  <div class="ripa-race tw-pb-8">
    <ripa-form-header title="Student" required subtitle="ยง999.224(a)(16)">
    </ripa-form-header>

    <v-container>
      <v-row no-gutters>
        <v-col cols="12" sm="12">
          <ripa-switch
            v-model="model.person.isStudent"
            label="K-12 Public School Student"
            :max-width="250"
            @input="handleInput"
          ></ripa-switch>
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script>
import RipaFormHeader from '@/components/molecules/RipaFormHeader'
import RipaFormMixin from '@/components/mixins/RipaFormMixin'
import RipaSwitch from '@/components/atoms/RipaSwitch'

export default {
  name: 'ripa-student',

  mixins: [RipaFormMixin],

  components: {
    RipaSwitch,
    RipaFormHeader,
  },

  data() {
    return {
      viewModel: this.loadModel(this.value),
    }
  },

  computed: {
    model: {
      get() {
        return this.viewModel
      },
    },
  },

  methods: {
    handleInput() {
      this.updateDisabilityModel()
      this.updateStopResultModel()
      this.$emit('input', this.viewModel)
    },

    updateDisabilityModel() {
      this.viewModel.person.anyDisabilities = false
    },

    updateStopResultModel() {
      if (!this.viewModel.person.isStudent) {
        this.viewModel.stopResult.actionsTakenDuringStop12 = false
        this.viewModel.stopResult.actionsTakenDuringStop13 = false
      }
    },
  },

  watch: {
    value(newVal) {
      this.viewModel = this.loadModel(newVal)
    },

    'viewModel.person.isStudent': {
      handler(newVal, oldVal) {
        if (oldVal !== newVal) {
          this.updateDisabilityModel()
          this.updateStopResultModel()
        }
      },
    },
  },

  props: {
    value: {
      type: Object,
      default: () => {},
    },
  },
}
</script>
