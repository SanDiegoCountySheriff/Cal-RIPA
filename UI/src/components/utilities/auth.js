import { PublicClientApplication } from '@azure/msal-browser'

export const doSignIn = async () => {
  await this.$msalInstance
    .loginPopup({})
    .then(() => {
      const myAccounts = this.$msalInstance.getAllAccounts()
      this.account = myAccounts[0]
      this.$emitter.emit('login', this.account)
    })
    .catch(error => {
      console.error(`error during authentication: ${error}`)
    })
}
