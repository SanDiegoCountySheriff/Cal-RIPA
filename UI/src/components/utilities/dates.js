export const formatDate = dateStr => {
  if (dateStr && dateStr.length > 0) {
    const str = dateStr.replace(/-/g, '')
    const year = str.substring(0, 4)
    const month = str.substring(4, 6)
    const day = str.substring(6, 8)
    return `${year}-${month}-${day}`
  }

  return ''
}
