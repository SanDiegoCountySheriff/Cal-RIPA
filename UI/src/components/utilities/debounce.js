export const debounce = (func, delay, immediate) => {
  let timerId
  return (...args) => {
    const boundFunc = func.bind(this, ...args)
    clearTimeout(timerId)
    if (immediate && !timerId) {
      boundFunc()
    }
    const calleeFunc = immediate
      ? () => {
          timerId = null
        }
      : boundFunc
    timerId = setTimeout(calleeFunc, delay)
  }
}
