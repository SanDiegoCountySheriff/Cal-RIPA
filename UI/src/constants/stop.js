export const SUBMISSION_STATUSES = [
  {
    text: 'Failed',
    value: 'Failed',
    editable: true,
  },
  {
    text: 'Submitted',
    value: 'Submitted',
    editable: false,
  },
  {
    text: 'Unsubmitted',
    value: 'Unsubmitted',
    editable: true,
  },
  {
    text: 'Resubmitted',
    value: 'Resubmitted',
    editable: false,
  },
]
