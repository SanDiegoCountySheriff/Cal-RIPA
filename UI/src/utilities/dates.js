import differenceInHours from 'date-fns/differenceInHours'
import { differenceInYears as diffInYears } from 'date-fns'
import format from 'date-fns/format'
import isValid from 'date-fns/isValid'
import endOfDay from 'date-fns/endOfDay'

export const formatDate = dateStr => {
  if (dateStr && dateStr.length > 0) {
    const str = dateStr.replace(/-/g, '')
    const year = str.substring(0, 4)
    const month = str.substring(4, 6)
    const day = str.substring(6, 8)
    return `${year}-${month}-${day}`
  }

  return ''
}

export const isValidDate = dateStr => {
  const date = parseDate(dateStr, '00:00')
  return isValid(date)
}

const parseDate = (dateStr, timeStr) => {
  return new Date(`${dateStr}T${timeStr}`)
}

export const dateNotInFuture = (dateStr, timeStr) => {
  const date = parseDate(dateStr, timeStr || '00:00')
  return new Date().getTime() >= date.getTime()
}

export const differenceInYears = date => {
  const diff = diffInYears(new Date(), new Date(date))
  return diff
}

export const dateWithinLastHours = (dateStr, timeStr, hours) => {
  const date = parseDate(dateStr, timeStr)
  const diff = differenceInHours(new Date(), date)
  return diff < hours
}

export const formatDateTime = (dateStr, timeStr) => {
  return new Date(`${dateStr}T${timeStr}`)
}

export const formatShortDate = dateStr => {
  const date = parseDate(dateStr, '00:00')
  return format(new Date(date), 'yyyy-MM-dd')
}

export const uniqueId = () => {
  return format(new Date(), 'yyyyMMddkkmm')
}

export const formatToIsoCurrentDate = () => {
  return format(endOfDay(new Date()), `yyyy-MM-dd'T'HH:mm:ss.SSSxxx`)
}

export const formatToIsoDate = date => {
  return format(endOfDay(date), `yyyy-MM-dd'T'HH:mm:ss.SSSxxx`)
}
