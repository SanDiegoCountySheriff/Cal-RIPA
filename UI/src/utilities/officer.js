export const getOfficerStartDate = () => {
  const startDate = localStorage.getItem('ripa_officer_start_date')
  return startDate || null
}

export const getOfficerYearsExperience = () => {
  const yearsExperience = localStorage.getItem('ripa_officer_years_experience')
  return +yearsExperience || null
}

export const getOfficerAssignment = () => {
  const assignment = localStorage.getItem('ripa_officer_assignment')
  return +assignment || null
}

export const getOfficerOtherType = () => {
  const otherType = localStorage.getItem('ripa_officer_other_type')
  return otherType || null
}
