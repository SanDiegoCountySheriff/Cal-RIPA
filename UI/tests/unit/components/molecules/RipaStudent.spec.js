import RipaStudent from '@/components/molecules/RipaStudent.vue'
import { shallowMount } from '@vue/test-utils'
import { defaultStop } from '@/utilities/stop'
import Vuetify from 'vuetify'

describe('Ripa Student', () => {
  let vuetify
  let wrapper
  let stop

  beforeEach(() => {
    vuetify = new Vuetify()
    stop = defaultStop()
  })

  afterEach(() => {
    wrapper.destroy()
  })

  const factory = propsData => {
    return shallowMount(RipaStudent, {
      vuetify,
      propsData: {
        ...propsData,
      },
    })
  }

  it('should match snapshot', () => {
    wrapper = factory({ value: stop })

    expect(wrapper.element).toMatchSnapshot()
  })
})
