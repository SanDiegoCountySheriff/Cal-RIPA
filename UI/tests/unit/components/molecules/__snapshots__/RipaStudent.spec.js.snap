// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Student should match snapshot 1`] = `
<div
  class="ripa-student tw-pb-4"
>
  <ripa-form-header-stub
    onopenstatute="[Function]"
    required="true"
    subtitle="ยง999.224(a)(16)"
    title="Student"
  />
   
  <v-container-stub
    tag="div"
  >
    <v-row-stub
      nogutters="true"
      tag="div"
    >
      <v-col-stub
        cols="12"
        sm="12"
        tag="div"
      >
        <ripa-switch-stub
          hint=""
          label="K-12 Public School Student"
          maxwidth="250"
          rules=""
        />
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;
